src/audio-view.js

```diff
@@ -0,0 +1,193 @@
+
+export default class AudioView {
+  constructor(store, config, dispatcher) {
+    this.store = store;
+    this.config = config;
+  }
+
+  // Loads all sounds, calls callback([err]) when done
+  load(callback) {
+    // Maps logical sound identifiers to filenames. We'll load these sounds next.
+    this.sounds = {
+      alone: {
+        ambient: new VCFSound({url: 'sounds/Alone_Mode/Pulse_Amb_Loop.wav', fadeIn: 3}),
+        handshake: 'sounds/Alone_Mode/Hand_Shake_01.wav'
+      },
+      mole: {
+        success: 'sounds/Game_01/G01_Success_01.wav',
+        failure: 'sounds/Game_01/G01_Negative_01.wav',
+        panels: [0,1,2].map(stripId => _.range(10).map(panelId => `sounds/Game_01/G01_LED_${("0"+(stripId*10+panelId+1)).slice(-2)}.wav`))
+      },
+//      disk: {
+//      },
+      simon: {
+        panels: [0,1,2].map(stripId => _.range(10).map(panelId => `sounds/Game_03/G03_LED_${("0"+(stripId*10+panelId+1)).slice(-2)}.wav`)),
+        success: 'sounds/Game_03/G03_Success_01.wav',
+        failure: 'sounds/Game_03/G03_Negative_01.wav',
+        show: 'sounds/Game_03/G03_Light_Show_01.wav'
+      }
+    };
+
+    // Traverse this.sounds and replace the filenames with valid sound objects.
+    this._promises = [];
+    this._traverse(this.sounds);
+
+    // _loadSound() will create promises. We call the callback once all promises resolve
+    console.log(`${this._promises.length} promises`);
+    Promise.all(this._promises)
+      // Don't listen to events until we've loaded all sounds
+      .then(() => this.store.on(SculptureStore.EVENT_CHANGE, this._handleChanges.bind(this)))
+      .then(() => callback(null))
+      .catch(callback.bind(null));
+  }
+
+  // Traverses sound config objects and replaces nodes with valid, loaded, sounds
+  _traverse(node) {
+    for (let key in node) {
+      const value = node[key];
+      let sound;
+      if (typeof value === 'string') sound = node[key] = new Sound({url: value});
+      else if (value instanceof Sound) sound = value;
+      if (sound) this._loadSound(sound);
+      else this._traverse(value);
+    }
+  }
+
+  // Returns a promise to provide a comple, usable sound.
+  _loadSound(sound) {
+    const promise = sound.load();
+    this._promises.push(promise);
+    return promise;
+  }
+
+  _handleChanges(changes) {
+    if (this._animating) {
+      return;
+    }
+
+    if (changes.currentGame === GAMES.HANDSHAKE) this.sounds.alone.ambient.play();
+
+    this._handleHandshakeChanges(changes);
+    this._handleStatusChanges(changes);
+    this._handleLightChanges(changes);
+  }
+
+  _handleHandshakeChanges(changes) {
+    if (changes.handshakes) {
+      // Did someone shake my hand?
+      if (changes.handshakes[this.config.username]) {
+        this.sounds.alone.ambient.stop();
+        this.sounds.alone.handshake.play();
+      }
+    }
+  }
+
+  _handleStatusChanges(changes) {
+    if (changes.status) {
+      let statusSounds;
+
+      if (this.store.isPlayingMoleGame) {
```

Comment:
I think you could reduce this code significantly by first getting the correct `this.sounds.*` property based on the current game and then getting `success` or `failure` based on the status. That would make it much more obvious what's going on here and would give us less code to maintain.